src/zope/schema/fieldproperty.py

```diff
@@ -28,11 +28,15 @@
 @interface.implementer(interfaces.IFieldUpdatedEvent)
 class FieldUpdatedEvent(object):
 
-    def __init__(self, inst, field, old_value, new_value):
-        self.inst = inst
+    def __init__(self, obj, field, old_value, new_value):
+        self.object = obj
         self.field = field
         self.old_value = old_value
         self.new_value = new_value
+        # The implementation used to differ from the interfaces in that it
+        # declared `self.inst` instead of `self.object`. Leave `self.inst`
+        # in place for backwards compat.
+        self.inst = obj
```

Comment:
I might suggest using a read/write property so that things stay in sync. It can be done in one line in the class definition:
```suggestion
        inst = property(lambda self: self.object, lambda self, new_value: setattr(self, 'object', new_value))
```

Adding support for `del` might or might not be useful.

Alternately, make `object` the read/write property in case people are checking `__dict__` for `inst`.

I don't know how important any of this is, but one nice thing about making `inst` a property: we could make it emit a `DeprecationWarning` before we decide to remove it.

Reply:
Done.

Reply:
Awesome, thanks!

One more minor ask: Could a test be updated to make sure that `event.inst = <something>` also means that `event.object is <something>`? Otherwise there's no way to know that writing to the property does the right thing, and not calling it probably drops our coverage numbers (actually, I don't think coverage catches unused lambdas' which is one reason why I've used them in the past 😉 )

Reply:
Done.